crates/chain-config/src/config/snapshot_metadata.rs

```diff
@@ -3,184 +3,97 @@ use std::path::{
-                if filepath.is_relative() {
-                    *filepath = dir.join(&filepath);
-                }
+            TableEncoding::Json { filepath } => {
+                *filepath = dir.join(&filepath);
             }
             #[cfg(feature = "parquet")]
-            StateEncoding::Parquet { filepaths, .. } => {
-                filepaths.prepend_relative_paths(dir)
+            TableEncoding::Parquet {
+                tables,
+                block_height,
+                da_block_height,
+                ..
+            } => {
+                for path in tables.values_mut() {
+                    *path = dir.join(&path);
+                }
+                *da_block_height = dir.join(&da_block_height);
+                *block_height = dir.join(&block_height);
             }
         }
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, PartialEq)]
 pub struct SnapshotMetadata {
-    metadata: Metadata,
+    pub chain_config: PathBuf,
+    pub table_encoding: TableEncoding,
 }
 
-#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, PartialEq)]
-struct Metadata {
-    chain_config: PathBuf,
-    state_encoding: StateEncoding,
-}
-impl Metadata {
+impl SnapshotMetadata {
     const METADATA_FILENAME: &'static str = "metadata.json";
-    fn new(state_encoding: StateEncoding) -> Self {
-        Self {
-            chain_config: CHAIN_CONFIG_FILENAME.into(),
-            state_encoding,
-        }
-    }
-    fn read(dir: &Path) -> anyhow::Result<Self> {
-        let path = dir.join(Self::METADATA_FILENAME);
+    pub fn read(dir: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let path = dir.as_ref().join(Self::METADATA_FILENAME);
         let file = std::fs::File::open(path)?;
-        Ok(serde_json::from_reader(&file)?)
-    }
+        let mut snapshot: Self = serde_json::from_reader(&file)?;
+        snapshot.prepend_path(dir.as_ref());
 
-    fn write(&self, dir: &Path) -> anyhow::Result<()> {
-        let path = dir.join(Self::METADATA_FILENAME);
-        let file = std::fs::File::create(path)?;
-        serde_json::to_writer_pretty(file, self)?;
-        Ok(())
+        Ok(snapshot)
     }
 
-    fn prepend_relative_paths(&mut self, dir: &Path) {
-        if self.chain_config.is_relative() {
-            self.chain_config = dir.join(&self.chain_config);
-        }
-        self.state_encoding.prepend_relative_paths(dir);
+    pub fn strip_prefix(&mut self, dir: &Path) -> anyhow::Result<&mut Self> {
+        self.chain_config = self.chain_config.strip_prefix(dir)?.to_owned();
+        self.table_encoding.strip_prefix(dir)?;
+        Ok(self)
     }
-}
 
-impl SnapshotMetadata {
-    pub fn chain_config(&self) -> &Path {
-        &self.metadata.chain_config
+    pub fn prepend_path(&mut self, dir: &Path) {
+        self.chain_config = dir.join(&self.chain_config);
+        self.table_encoding.prepend_path(dir);
     }
```

Comment:
It seems these methods can be private and we can call them only on `read` and `write`.

Reply:
Also, I would prefer to remove these methods completely and put the underlying logic directly into the constructor/flush functions. But it is up to you=)